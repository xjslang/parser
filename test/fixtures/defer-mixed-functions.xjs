function outerFunction() {
  console.log('outer start')
  defer {
    console.log('outer defer')
  }
  
  const innerArrow = () => {
    console.log('inner arrow start')
    defer {
      console.log('inner arrow defer')
    }
    console.log('inner arrow end')
  }
  
  const innerExpression = function() {
    console.log('inner expression start')
    defer {
      console.log('inner expression defer')
    }
    console.log('inner expression end')
  }
  
  innerArrow()
  innerExpression()
  console.log('outer end')
}

outerFunction()
