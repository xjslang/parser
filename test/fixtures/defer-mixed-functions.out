outer start
inner arrow start
inner arrow end
inner arrow defer
inner expression start
inner expression end
inner expression defer
outer end
outer defer
